### ISSUE TITLE
Alchemical Golems don't consider other containers if Essentia Condenser is in target

### ISSUE THREAD

**Issue by monroth** [GTNewHorizons/GT-New-Horizons-Modpack#15222]
### Your GTNH Discord Username

Monroth

### Your Pack Version

2.5.1

### Your Server

Private

### Java Version

Java 17

### Type of Server

Vanilla Forge

### Your Expectation

Put golem with "alchemy" core to put essentia into Essentia condenser and oblivion jar. Expected it to put essentia into condenser if it has space, and into oblivion jar if it does not (like it behaves with jars and reservoirs).

### The Reality

When condenser became full golem stopped to carry essentia, jamming the network.

### Your Proposal

Make so golems work with EC in the same way as with other essentia containers - leftovers should be brought to other places.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by the-bukkit** [comment]
I also have this issue on version 2.5.1

The alchemy golem is supposed to move essentia from the left reservoir to the condenser, or if the condenser is full to the reservoir on the right.

However since the condenser is full of that type (aqua essentia) it's stuck and not moving on to the reservoir.

![2023-12-31_00 05 08](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/23724545/34d47567-86df-46a6-9fe7-1f9eabd83974)

**Comment by the-bukkit** [comment]
I've done some more digging and found that Thaumcraft itself may be the issue.

In the **"shouldExecute()"** method of the alchemy golem **(AIEssentiaEmpty.java)** it calls **"findJarWithRoom(...)" (from GolemHelper.java)** to find a container that the essentia can be inserted into.
With jars it checks if there is available space, unlike the essentia reservoir or with other IEssentiaTransport objects like the essentia condenser from gadomancy.
The reason why reservoirs don't also have this problem is that they reduce their suction to 0 when full, but since the essentia condenser has seperate _storages_ for different types of essentia it doesn't reduce it's suction.

### TL;DR
Thaumcraft's alchemy golems do not check for available space in that type of storage and the condenser doesn't / can't lower it's suction value.
